natlib/svcd.c

```diff
@@ -185,3 +186,50 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+static int svcd_subscribe(lua_State* L)
+{
+    if (lua_gettop(L) != 4) return luaL_error(L, "Expected (targetip, svcid, attrid, on_notify)");
+     
+    lua_getglobal(L, "SVCD");
+    lua_getfield(L, -1, "ivkid");
+     
+    uint16_t ivkid = luaL_checkinteger(L, -1);
+    lua_pop(L, 1);
+
+    uint16_t new_ivkid = ivkid + 1;
+    if (new_ivkid > 65535) {
+        new_ivkid = 0;
+    }
+
+    lua_pushnumber(L, new_ivkid);
+    lua_setfield(L, -1, "ivkid");
+     
+    lua_getfield(L, -1, "oursubs");
+    // put value on the stack
+    lua_pushinteger(L, ivkid);
+    lua_pushvalue(L, 4); // on_notify
+    lua_settable(L, -3);
+    lua_pop(L, 2);
+     
+     
+    lua_pushlightfunction(L, libstorm_net_sendto);
+     
+    lua_getglobal(L, "SVCD");
```

Comment:
You have already gotten SVCD in the beginning of the function. If you know that you might need it again, why not just keeping it in the memory and recycle it here?

Untitled